### ISSUE TITLE
datamodels.open fails to read x1dints for TSO

### ISSUE THREAD

**Issue by zkbt** [spacetelescope/jwst#6113]
Hi all!

(Sorry if I'm posting an issue you already know about, or that I've stumbled onto something that's still in-progress. Please feel free to ignore!) 

I'm working with some simulated data for the ers-transit team (including MIRI LRS simulations from MIRISimand NIRCam simulations from mirage). We have some `x1dints` files that come from processing the simulated images with the pipeline, and I was trying to open them to learn more about how datamodels work. 

I created a clean conda environment and installed `jwst` into it. I ran this with both the basic `pip install jwst` version which worked out to be version 1.2.0, and with the development version straight from github, which was `1.2.1.dev15+gb5fe85dd`.  

I tried to open an `x1dints` with the following snippet:
```
from jwst import datamodels
filename = '/Users/zach/Data/JWST/NIRCam/Stage 2 Output/jw00042001001_01101_00002-seg001_nrca5_x1dints.fits'
x1d = datamodels.open(filename)
```

However, I got the following error:
```
~/opt/anaconda3/envs/ers-transit/lib/python3.9/site-packages/stdatamodels/util.py in gentle_asarray(a, dtype)
     67                 out_dtype.names = in_dtype.names
     68             else:
---> 69                 raise ValueError(
     70                     "Column names don't match schema. "
     71                     "Schema has {0}. Data has {1}".format(

ValueError: Column names don't match schema. Schema has {'flux_var_rnoise', 'flux_var_poisson', 'bkgd_var_rnoise', 'flux_error', 'sb_var_poisson', 'bkgd_error', 'flux_var_flat', 'bkgd_var_poisson', 'sb_var_flat', 'bkgd_var_flat', 'sb_var_rnoise'}. Data has {'berror', 'error'}
```
I got the same error on simulated MIRI, NIRCam, and NIRSpec data. All of these were TSO observations.

Talking with some other folks on the team (Thomas Beatty), we identified that version 0.18.0 of `jwst` can open our simulated `x1dints` files just fine, but the current version 1.2.0 cannot (nor can the development version). When I reverted to 0.18.0 on my own computer, I was able to read the files. 

I don't know if this is a problem with datamodel trying to read a file, or with the pipeline outputs for x1dints not having yet caught up to changes to the datamodel. Either way, I hope it's useful that I let you know.

Cheers,
Zach

**Comment by tapastro** [comment]
Hi Zach,

The more modern releases of jwst that you've tried include an update to all spectral data products, containing propagated uncertainty columns - these are the schema column names provided above. This update also changed the name of the flux error and background error column names to improve clarity. These columns will be calculated and provided during all future runs of the Level 2/3 pipeline. The simulated data should probably be brought up to date with this standard, but we (@eslavich) also created a data migration script that should address this issue: https://github.com/spacetelescope/jwst/pull/6055

**Comment by zkbt** [comment]
Awesome, thanks so much! 

That `migrate_data` script totally worked on our simulated MIRI data. I was then able to read in the files with `datamodels.open`, and they showed up as `MultiSpecModel` with wavelengths and fluxes populated as I expected. The script didn't work on our NIRCam or NIRSpec simulations (saying `Migration for ... CAL_VER 1.1.1 not implemented`); I assume this is just that they must have been made with a slightly different version of the pipeline. 

This is a niche thing, and I don't want to trouble you with fixing all these temporary corner cases, so please don't worry about fixing the NIRCam/NIRSpec issue for our sakes. We have plenty to go off of with MIRI for now! (And, I assume, could just rerun the extraction on the simulated data to produce our own more modern x1dints files.)

Thanks!!

**Comment by eslavich** [comment]
The NIRCam and NIRSpec files must have been created with a dev version of the jwst package?  We never released a 1.1.1, anyway.  The script tries to be careful not to mangle a file when it doesn't understand how it was created, and since it doesn't know whether the dev version occurred before or after the change to the x1d format, it makes the conservative choice and takes no action.

Since you know you have files in the old format, you can trick the script by setting the CAL_VER keyword to 1.1.0.
